app/routes/_gcn.synonyms/synonyms.server.ts

```diff
@@ -40,3 +40,61 @@ export async function createSynonyms(...synonymousEventIds: string[]) {
 
   return uuid
 }
+
+/*
+ * If an eventId already has a synonym and is passed in, it will unlink the
+ * eventId from the old synonym and the only remaining link will be to the
+ * new synonym.
+ *
+ * BatchWriteItem has a limit of 25 items, so the user may not add and/or remove
+ *  more than 25 synonyms at a time.
+ */
+export async function putSynonyms({
+  uuid,
+  additions,
+  subtractions,
+}: {
+  uuid: string
+  additions?: string[]
+  subtractions?: string[]
+}) {
+  const db = await tables()
+  const client = db._doc as unknown as DynamoDBDocument
+  const writes = []
+  if (!subtractions && !additions) {
+    const { Items } = await db.synonym.query({
+      IndexName: 'synonymsByUuid',
+      KeyConditionExpression: 'uuid = :uuid',
+      ExpressionAttributeValues: {
+        ':uuid': uuid,
+      },
+    })
+    return Items
+  }
+  if (subtractions) {
+    const subtraction_writes = subtractions
+      .filter((eventId) => Boolean(eventId))
```

Comment:
This checks that `eventId` is not an empty string. While it would be unusual if it was an empty string, nothing would actually break. Is this validation necessary, and is it this function's responsibility?

```suggestion
```